desafio_spring/src/main/java/meli/bootcamp/desafio_spring/services/UserService.java

```diff
@@ -63,6 +65,27 @@ public void followSeller(Long userId, Long sellerId){
         user.followSeller(seller);
         seller.addFollower(user);
 
+        try{
+            this.userRepository.save(user);
+            this.sellerRepository.save(seller);
+        } catch (DataIntegrityViolationException e){
+            throw new DuplicatedResouceException("The user with id " + userId + " already follows the seller " + sellerId);
+        }
+    }
+
+    public void unFollowSeller(Long userId, Long sellerId){
+
+        User user = this.userRepository.findById(userId).orElseThrow(() ->
+                new ResourceNotFoundException("The user with id " + userId + " doesn't exist.")
+        );
+
+        Seller seller = this.sellerRepository.findById(sellerId).orElseThrow(() ->
+                new ResourceNotFoundException("The seller with id " + sellerId + " doesn't exist.")
+        );
+
+        user.unfollowSeller(seller);
```

Comment:
Como sempre que o user for dar unfollow no seller é necessário remover a referência da entidade do seller, acho que chamar o seller.removeFollower dentro do método unfollowSeller do User.

Reply:
Muito bem levantado, adotei essa dica e já implementei! muito obrigado!